paddle/phi/api/lib/tensor.cc

```diff
@@ -217,20 +222,28 @@ T *Tensor::mutable_data(const Place &place) {
   return nullptr;
 }
 
-template PADDLE_API float *Tensor::mutable_data<float>(const Place &place);
-template PADDLE_API double *Tensor::mutable_data<double>(const Place &place);
-template PADDLE_API int64_t *Tensor::mutable_data<int64_t>(const Place &place);
-template PADDLE_API int32_t *Tensor::mutable_data<int32_t>(const Place &place);
-template PADDLE_API uint8_t *Tensor::mutable_data<uint8_t>(const Place &place);
+template PADDLE_API bool *Tensor::mutable_data<bool>(const Place &place);
 template PADDLE_API int8_t *Tensor::mutable_data<int8_t>(const Place &place);
+template PADDLE_API uint8_t *Tensor::mutable_data<uint8_t>(const Place &place);
 template PADDLE_API int16_t *Tensor::mutable_data<int16_t>(const Place &place);
-template PADDLE_API bool *Tensor::mutable_data<bool>(const Place &place);
+template PADDLE_API uint16_t *Tensor::mutable_data<uint16_t>(
+    const Place &place);
+template PADDLE_API int32_t *Tensor::mutable_data<int32_t>(const Place &place);
+template PADDLE_API uint32_t *Tensor::mutable_data<uint32_t>(
+    const Place &place);
+template PADDLE_API int64_t *Tensor::mutable_data<int64_t>(const Place &place);
+template PADDLE_API uint64_t *Tensor::mutable_data<uint64_t>(
+    const Place &place);
```

Comment:
<img width="625" alt="image" src="https://user-images.githubusercontent.com/13048366/217132179-be2ffc22-c8f6-4e1e-b40a-bd47f2c5691a.png">
输入place参数的mutable_data接口计划不再支持了，这里是否还需要再增加支持的类型？

Reply:
个人理解此处可以支持：
1. 增加数据类型不影响未来废弃此接口
2. 支持的数据类型如果和 `data()` 函数不一致，会增加开发者的编码成本

In my opinion, we can support more datatypes despite this is a deprecated method:
1. Support more datatypes will not affect deprecating this method in the future.
2. It will cost developers more time to debug if the datatype supported by this method is different of functions like `data()` /

Reply:
这种情况是希望用户最好不再新增对这个接口的使用，用户每新增一次调用都会加大未来废弃接口的负担，如果降低了开发者的编码成本，可能反而会让用户更容易继续用这个接口

Reply:
这里不建议加，里面有warning的，按理说我们已经可以移除了

Reply:
已修复，此处未支持更多的数据类型

Done, this function does not support more datatypes.